src/templates/post.js

```diff
@@ -82,21 +83,59 @@ const BlogPost = ({ data }) => {
     };
   }, []);
 
+
   const handleScroll = () => {
     const moreFromXlSize = refMoreFrom?.current?.clientHeight || 0;
-     const isInbottom = Math.ceil(window.innerHeight + window.scrollY + moreFromXlSize + 200) >= document.documentElement.scrollHeight;
-     if (isInbottom) {
-       setDisappearSocial(true);
-     } else {
-       if (!disappearSocial) {
-         setDisappearSocial(false);
-       }
-     };
-   }; 
-  
+    const isInbottom = Math.ceil(window.innerHeight + window.scrollY + moreFromXlSize + 200) >= document.documentElement.scrollHeight;
+    const isInTop = document.documentElement.scrollTop < 800;
+    
+    if (isInbottom) {
+      setDisappearSocial(true);
+    } else {
+      if (!disappearSocial) {
+        setDisappearSocial(false);
+      }
+    };
+    if (isInTop || isInbottom) {
+      setDisappearIndex(true);
+    } else {
+      if (!disappearIndex) {
+        setDisappearIndex(false);
+      }
+    };
+  };
+
+  const getTitles = () => {
+    if (typeof window !== 'undefined' && typeof window.document !== "undefined") {
+      const postContainer = document.getElementById('postContainer');
+      if (postContainer !== null) {
+        const elementList = Array.from(postContainer.childNodes);
+        const titlesList = findTitles(elementList);
+        return (
+        <div className={classnames({[styles.disappearIndex]: disappearIndex},styles.indexSubContainer)}>
+          {titlesList.map((title) => 
+          <a href={"#" + title.id} key={title.id} onClick={() => setSelectLink(title.id)} 
+            className={classnames(
+              {[styles.selectedLink]:title.id === selectLink}, 
+              styles.links, 
+          )}>
+            {title.id === selectLink && <PinkCircle className={styles.pinkCircle}/>}
+            {title.innerHTML.trim().split(/\s+/).length > 10 ? title.innerHTML.slice(0, 55) + "..." : title.innerHTML}
```

Comment:
What do you try to do here?

Reply:
I want to cut the text in ten words, also, I want to show at the final of the cut three points.  The text can't be in three lines. For this reason, I cut it.

Reply:
Why don't use split by space and get the first 10 places in the array? Also, what happens if the last character is a comma? for example:` hi bros!, etc etc`. Do you put: `hi bros!,....` ?

Reply:
If the last character is a comma, I think the final three dots will be next to it. It is the same with the other symbols, if you think that will have to change, I can ask Isa for an opinion.

Reply:
Done